databases/week3/database_week3.sql

```diff
@@ -0,0 +1,267 @@
+INSERT INTO reservation (id, 
+	 number_of_guests,
+     meal_id,
+	 created_date) 
+ VALUE(4, 22, 6, '2020-06-27 18:15:59');
+ 
+-- Get a reservation with any id, 
+SELECT * FROM reservation 
+WHERE reservation.id = 4;
+
+-- Update a reservation with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE reservation SET reservation.number_of_guests = 10, reservation.meal_id = 5
+WHERE id = 4;
+
+-- Delete a reservation with any id, 
+DELETE FROM reservation WHERE reservation.id = 4;
+
+-- Review
+-- Get all reviews
+SELECT * FROM review;
+
+-- Add a new review
+INSERT INTO review (id, title,
+     description, meal_id, 
+     stars, created_date) 
+ VALUE(3, 'cool one', 
+	  'Really fantastic mocoronni meal',
+	 6, 4, '2020-07-03 10:15:28');
+INSERT INTO review (id, title,
+     description, meal_id, 
+     stars, created_date) 
+ VALUE(4, 'fantastic', 
+	  'It is exceptionally tasty shiwarama pizza',
+	 5, 5, '2020-06-30 16:25:00');
+
+-- Get a review with any id
+SELECT * FROM review WHERE review.id = 3;
+
+-- Update a review with any id. Update any attribute fx the title or multiple attributes
+UPDATE review SET review.stars = 2, review.title = 'Not bad'
+WHERE review.id = 3;
+
+-- Delete a review with any id, 
+ DELETE FROM review WHERE review.id = 3;
+
+-- Additional queries adding some reservations and reviews 
+INSERT INTO reservation (id, 
+	 number_of_guests,
+     meal_id,
+	 created_date) 
+ VALUE(5, 26, 5, '2020-06-21 20:05:00');
+ INSERT INTO review (id, title,
+     description, meal_id, 
+     stars, created_date) 
+ VALUE(5, 'Green', 
+	  'It was special vegetables meal',
+	 3, 4, '2020-06-28 19:50:30');
+INSERT INTO review (id, title,
+     description, meal_id, 
+     stars, created_date) 
+ VALUE(3, 'sandwich', 
+	  'Tasty open sandwich',
+	 2, 4, '2020-06-19 13:15:00');
+
+-- Functionality queries
+-- Get meals that has a price smaller than a specific price 
+SELECT meal.id, meal.title, meal.price FROM meal
+WHERE meal.price < 200;
+ 
+-- Get meals that still has available reservations
+SELECT meal.id AS meal_id, meal.title, meal.max_reservations 
+FROM meal
+LEFT JOIN reservation ON meal.id = reservation.meal_id
+WHERE (meal.max_reservations - reservation.number_of_guests) > 0
+GROUP BY meal.id;
+ 
+-- Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+SELECT * FROM meal WHERE meal.title LIKE '%pizza' OR meal.title LIKE 'pizza%';
+
+-- Get meals that has been created between two dates
+SELECT * FROM meal WHERE meal.created_date 
+BETWEEN date('2020-06-25 20:30:20') AND date('2020-06-27 10:30:20');
+    
+-- Get only specific number of meals fx return only 5 meals
+SELECT * FROM meal  LIMIT 3;
+ 
+-- Get the meals that have good reviews(asuming meals with greater than four stars as good)
+SELECT meal.id AS meal_id, meal.title, review.stars   
+FROM meal
+INNER JOIN review ON meal.id = review.meal_id
+WHERE review.stars > 3
+ORDER BY meal.id;
+
```

Comment:
In this case you need to get the average of the all the reviews which a meal has received in it's entirety
```

SELECT Avg(r.stars) AS avg_review,
       m.title
FROM   meal m,
       review r
WHERE  m.id = r.meal_id  and r.stars > 3
GROUP  BY r.meal_id
ORDER  BY Avg(r.stars) DESC;  

```